### ISSUE TITLE
Request feature: Triggers

### ISSUE THREAD

**Issue by sonhoangnz** [olirice/alembic_utils#18]
Hello

Thanks, for the great library. Now I can use alembic_utils for creating/updating Procedures and Views.

My request/question is can we extend this library for creating/updating Triggers ?

Many thanks

**Comment by olirice** [comment]
Hi,

Yes, I am interested in supporting triggers. I created a branch for them this morning

You can install that branch with 
```
pip install git+https://github.com/olirice/alembic_utils.git@pg_trigger
```

General usage is 

```python
from alembic_utils.pg_trigger import PGTrigger

trigger = PGTrigger(
    schema="public",
    signature="lower_account_email",
    definition="""
        BEFORE INSERT ON public.account
        FOR EACH ROW EXECUTE FUNCTION public.downcase_email()
    """,
)

# OR

trigger = (
    PGTrigger
    .from_sql(
        """create trigger lower_account_email
        BEFORE INSERT ON public.account
        FOR EACH ROW EXECUTE FUNCTION public.downcase_email()
    """)
)
```  

the should be registered with `register_entities` just like views and functions.

It has full test coverage but may not handle every edge case perfectly. Please try it out by installing the branch and let me know if you run into any difficulty.

Once I hear back from you and address any issues you find I'll ping a few others to review/test and get it released on pypi

**Comment by olirice** [comment]
Would you mind telling me which company/project you're using `alembic_utils` with?
I'd like to start keeping track to better understand which use cases should get the most attention. Thanks!

**Comment by sonhoangnz** [comment]
Thanks, Olirice, for very fast response :+1: 

I am working for catalyst.co.nz and need to use Alembic migration tool to make sure dbs (prod, dev) and models are consistent for an internal project. The project db has a number of functions/views and triggers, and alembic_utils is helping us well.

Is there any change the library can work with PG v.10. For migration, there are still many apps are using old PG.

Many thanks, again

**Comment by olirice** [comment]
Thanks for letting me know!

if you pull the latest version of the pg_trigger branch with the command posted above, `alembic_utils.pg_trigger.PGTrigger`
will work with postgres 10 and 11

Please note that while postgres 10 "works" in this case, the project only officially supports 11 and above. The test suite does run against pg 10 in CI so I know if something breaks but if it causes difficulty it may be dropped any without a deprecation cycle.

Let me know if you run into any trouble

**Comment by olirice** [comment]
@daviddaskell pinging b/c you expressed interest in trigger support previously

**Comment by olirice** [comment]
Available in pypi release 0.2.11
